dispatcher/backend/src/routes/utils.py

```diff
@@ -1,34 +1,92 @@
-    """replaces (in-place) all occurences of secrets in flags/commands with stars"""
+    """replaces or removes (in-place) all occurences of secrets"""
+    remove_secret_flags(response)
+    remove_upload_secrets(response)
+
 
-    if "config" not in response:
+def remove_secret_flags(response: dict):
+    """replaces (in-place) all occurences of secret flags with stars"""
+
+    # we need flags to retrieve the secret value and replace it where we find it
+    # in commands
+    if "config" not in response or "flags" not in response["config"]:
         return
 
     schema = ScheduleConfigSchema.get_offliner_schema(
         response["config"]["task_name"]
     )().to_desc()
-    fields = [flag["data_key"] for flag in schema if flag.get("secret", False)]
+    fields = [field["data_key"] for field in schema if field.get("secret", False)]
+
+    flags = response["config"]["flags"]
 
+    # look after each fields of schema in flags
     for field in fields:
-        flags = response["config"]["flags"]
-        command = response["config"].get("command", [])
-        if field in flags:
-            flags[field] = SECRET_REPLACEMENT
-            try:
-                index = command.index(f'--{field}="{flags[field]}"')
-                command[index] = f'--{field}="{SECRET_REPLACEMENT}"'
-            except ValueError:
-                continue
-            if response.get("container"):
-                response["container"]["command"][
-                    index
-                ] = f'--{field}="{SECRET_REPLACEMENT}"'
-        if command:
-            response["config"]["str_command"] = build_str_command(
-                response["config"]["command"]
+        if field not in flags:
+            continue
+
+        secret_flag_value = flags[field]
+
+        flags[field] = SECRET_REPLACEMENT
+
+        for separator in ["", "'", '"']:
+            secret_command_value = (
+                f"--{field}={separator}{secret_flag_value}{separator}"
+            )
+            remove_secret_flag_from_command(
+                response, "config", field, secret_command_value
+            )
+
+            if "container" in response:
+                remove_secret_flag_from_command(
+                    response, "container", field, secret_command_value
+                )
+
+    # rebuild str_command from command
+    if "str_command" in response["config"]:
+        response["config"]["str_command"] = build_str_command(
+            response["config"]["command"]
+        )
+
+
+def remove_secret_flag_from_command(
+    response: dict, root_key_value: str, field_name: str, secret_value: str
+):
+    if "command" not in response[root_key_value]:
+        return
+    if secret_value not in response[root_key_value]["command"]:
+        return
+    index = response[root_key_value]["command"].index(secret_value)
+    response[root_key_value]["command"][
+        index
+    ] = f'--{field_name}="{SECRET_REPLACEMENT}"'
```

Comment:
If `index` is the position in list and contains `secret_value`, then assigning `--{field-name}="secret"` to `command[index]` will result in a list with the following I think

```py
command [ 
    ...
    '--field-name',
    '--field-name="secret"',
    …
]
```

Reply:
You're right ; it only applies to command which has this format.